os/kernel/binary_manager/binary_manager_getinfo.c

```diff
@@ -297,3 +217,53 @@ void binary_manager_get_info_all(int requester_pid)
 #endif
 	binary_manager_send_response(q_name, &response_msg, sizeof(binmgr_getinfo_all_response_t));
 }
+
+int binary_manager_get_download_path(int requester_pid, char *bin_name)
+{
+#ifdef CONFIG_APP_BINARY_SEPARATION
+	int bin_idx;
+#endif
+	binmgr_kinfo_t *kerinfo;
+	char q_name[BIN_PRIVMQ_LEN];
+	binmgr_getdownloadpath_response_t response_msg;
+
+	if (requester_pid < 0 || bin_name == NULL) {
+		bmdbg("Invalid data : pid %d name %s \n", requester_pid, bin_name);
+		response_msg.result = BINMGR_INVALID_PARAM;
+		goto send_result;
+	}
+
+	/* If it is kernel, Return the devname of inactive kernel partition */
+	if (!strncmp("kernel", bin_name, BIN_NAME_MAX)) {
+		kerinfo = binary_manager_get_kdata();
+		if (kerinfo->part_count > 1) {
+			response_msg.result = BINMGR_OK;
+			snprintf(response_msg.binpath, BINARY_PATH_LEN, BINMGR_DEVNAME_FMT, kerinfo->part_info[kerinfo->inuse_idx ^ 1].devnum);
+		} else {
+			response_msg.result = BINMGR_NOT_FOUND;
+		}
+		goto send_result;
+	}
+
+#ifdef CONFIG_APP_BINARY_SEPARATION
+	bin_idx = binary_manager_get_index_with_name(bin_name);
```

Comment:
How about like below?
```
tcb->group->tg_binidx
``
Maybe it is much faster than your code.

Reply:
It can't get the tcb from binary name here.
So I get the `bin_idx` from `bin_name` with `binary_manager_get_index_with_name`